src/main/java/io/aiven/kafka/connect/transforms/HashField.java

```diff
@@ -0,0 +1,192 @@
+        final SchemaAndValue schemaAndValue = getSchemaAndValue(record);
+        if (schemaAndValue.schema() == null) {
+            throw new DataException(dataPlace() + " schema can't be null: " + record);
+        }
+
+        final Optional<String> newValue;
+        if (config.fieldName().isPresent()) {
+            newValue = getNewValueForNamedField(
+                    record.toString(), schemaAndValue.schema(), schemaAndValue.value(), config.fieldName().get());
+        } else {
+            newValue = getNewValueWithoutFieldName(
+                    record.toString(), schemaAndValue.schema(), schemaAndValue.value());
+        }
+
+        if (newValue.isPresent()) {
+            return record.newRecord(
+                    record.topic(),
+                    record.kafkaPartition(),
+                    record.keySchema(),
+                    record.key(),
+                    record.valueSchema(),
+                    newValue.get(),
+                    record.timestamp(),
+                    record.headers()
+            );
+        } else {
+            return record;
+        }
+    }
+
+    @Override
+    public void close() {
+    }
+
+    protected abstract String dataPlace();
+
+    protected abstract SchemaAndValue getSchemaAndValue(final R record);
+
+    private Optional<String> getNewValueForNamedField(final String recordStr,
+                                                      final Schema schema,
+                                                      final Object value,
+                                                      final String fieldName) {
+        if (Schema.Type.STRUCT != schema.type()) {
+            throw new DataException(dataPlace() + " schema type must be STRUCT if field name is specified: "
+                    + recordStr);
+        }
+
+        if (value == null) {
+            throw new DataException(dataPlace() + " can't be null if field name is specified: " + recordStr);
+        }
+
+        final Field field = schema.field(fieldName);
+        if (field == null) {
+            return Optional.empty();
+        }
+
+        if (!SUPPORTED_TYPES_TO_CONVERT_FROM.contains(field.schema().type())) {
+            throw new DataException(fieldName + " schema type in " + dataPlace()
+                    + " must be " + SUPPORTED_TYPES_TO_CONVERT_FROM
+                    + ": " + recordStr);
+        }
+
+        final Struct struct = (Struct) value;
+
+        final Optional<String> result = Optional.ofNullable(struct.get(fieldName))
+                .map(Object::toString).flatMap(s -> config.hashFunction().map(alg -> hashString(alg, s)));
+        if (result.isPresent() && !result.get().equals("")) {
+            return result;
+        } else {
+            return Optional.empty();
+        }
+    }
+
+    static final String hashString(final String hashAlg, final String input) {
+        try {
+            final MessageDigest md = MessageDigest.getInstance(hashAlg);
```

Comment:
Could you please initialize this somewhere in `configure`? `MessageDigest.getInstance` might be costly enough to not want to call it on each message.

Reply:
Yeah that’s a good call out I’ll make that change.